packages/server/routes/api/guilds.js

```diff
@@ -26,29 +27,179 @@ router.get(
   AuthController.authenticate,
   async (request, response) => {
     try {
-      const { guildId } = request.params;
+      const guildId = request.params.guildId;
+      const guild = await GuildController.getGuild(guildId);
 
-      if (guildId === undefined) {
-        return response.status(400).json({
+      return response.status(200).json({
+        status: "success",
+        data: {
+          guild,
+        },
+      });
+    } catch (error) {
+      // Note: Keep this here until Prisma NotFoundError is fully deprecated with the future version of the client, as of now, it will not reach the P2025 case in the conditional below.
+      if (error.code === "P2025") {
+        return response.status(404).json({
           status: "fail",
           data: {
-            guildId: "Guild ID not provided",
+            guildId: `No guild with an ID of ${request.params.guildId} could be found.`,
           },
         });
       }
 
-      const guild = await GuildController.getGuild(guildId);
-      response.status(200).json({
+      if (error.name === "PrismaClientKnownRequestError") {
+        switch (error.code) {
+          case "P2023":
+            return response.status(400).json({
+              status: "fail",
+              data: {
+                guildId: "Provided guild ID is not a valid UUID.",
+              },
+            });
+
+          case "P2025":
+            return response.status(404).json({
+              status: "fail",
+              data: {
+                guildId: `No guild with an ID of ${request.params.guildId} could be found.`,
+              },
+            });
+
+          default:
+            return response.status(500).json({
+              status: "fail",
+              errorName: error.name,
+              errorMessage: error.message,
+            });
```

Comment:
This default case is unnecessary since it's just a copy of the default error handler and this goes for the other routes with switch statements in error handling too. Keep this if statement but remove the else block wrapping the default error response below this line. Since your switch cases have return statements, the default error handler below won't run if the switch cases are executed.